src/libs/ReportActionsUtils.ts

```diff
@@ -374,14 +374,21 @@ function shouldIgnoreGap(currentReportAction: ReportAction | undefined, nextRepo
  * Returns a sorted and filtered list of report actions from a report and it's associated child
  * transaction thread report in order to correctly display reportActions from both reports in the one-transaction report view.
  */
-function getCombinedReportActions(reportActions: ReportAction[], transactionThreadReportActions: ReportAction[], reportID?: string): ReportAction[] {
-    if (isEmptyObject(transactionThreadReportActions)) {
+function getCombinedReportActions(
+    reportActions: ReportAction[],
+    transactionThreadReportID: string | null,
+    transactionThreadReportActions: ReportAction[],
+    reportID?: string,
+): ReportAction[] {
+    // There is a chance that transactionThreadReportActions is not loaded yet.
+    // We only check for emptiness when there are multiple IOUs,
+    // this will prevent the sub-report preview from being displayed when there is only one IOU.
+    if (isEmptyObject(transactionThreadReportActions) && _.isEmpty(transactionThreadReportID)) {
```

Comment:
```suggestion
    if (_.isEmpty(transactionThreadReportID)) {
```

Reply:
@Pujan92, could you provide further clarification on this suggestion?

Reply:
I understand, I think that makes sense. The PR has been updated.